chia/full_node/weight_proof.py

```diff
@@ -1734,3 +1652,76 @@ def _validate_vdf_batch(
             return False
 
     return True
+
+
+async def validate_weight_proof_inner(
+    constants,
+    executor,
+    shutdown_file_name,
+    num_processes,
+    weight_proof: WeightProof,
+    summaries: List[SubEpochSummary],
+    sub_epoch_weight_list: List[uint128],
+    skip_segment_validation: bool,
+    validate_from: int,
+) -> Tuple[bool, List[BlockRecord]]:
+    assert len(weight_proof.sub_epochs) > 0
+    if len(weight_proof.sub_epochs) == 0:
+        return False, []
+
+    peak_height = weight_proof.recent_chain_data[-1].reward_chain_block.height
+    log.info(f"validate weight proof peak height {peak_height}")
+    seed = summaries[-2].get_hash()
+    rng = random.Random(seed)
```

Comment:
we depend on both peers using the same method so we depend on pythons implementation rn i raised this issue at the time (pre mainnet) and was told it is OK since we dont plan on multi client and we can change it without forking

Reply:
I just wouldn't expect python to guarantee stability on this